### PR TITLE
TF: TFMarianMTModel final logits bias as a layer

### LINKED ISSUE #18802: `load_tf_weights` doesn't handle the weights added to the TF models at the top level
### System Info

- `transformers` version: 4.22.0.dev0
- Platform: Windows-10-10.0.22000-SP0
- Python version: 3.9.11
- Huggingface_hub version: 0.8.1
- PyTorch version (GPU?): 1.12.1+cu113 (True)
- Tensorflow version (GPU?): 2.9.1 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

@gante

### Reproduction

(TF)MarianMTModel has weights `final_logits_bias` added at the top-level (i.e. not under any layer)

https://github.com/huggingface/transformers/blob/5f06a09b9f3f05b4860f11bbbe22861923b49d81/src/transformers/models/marian/modeling_tf_marian.py#L1287

However, the method `load_tf_weights` only handle weights under some layers

https://github.com/huggingface/transformers/blob/5f06a09b9f3f05b4860f11bbbe22861923b49d81/src/transformers/modeling_tf_utils.py#L850

This causes problem when we load TF checkpoints for `TFMarianMTModel`, i.e. `final_logits_bias` is not loaded.

```python
from transformers import MarianMTModel, TFMarianMTModel
model_name = "Helsinki-NLP/opus-mt-en-ROMANCE"

pt_model = MarianMTModel.from_pretrained(model_name)
tf_model_from_pt = TFMarianMTModel.from_pretrained(model_name, from_pt=True)
tf_model = TFMarianMTModel.from_pretrained(model_name, from_pt=False)

# Only has `TFMarianMainLayer` in `layers`
print(tf_model.layers)

print(pt_model.final_logits_bias.numpy())
print(tf_model_from_pt.final_logits_bias.numpy())
print(tf_model.final_logits_bias.numpy())
```

Outputs:

```bash
[<transformers.models.marian.modeling_tf_marian.TFMarianMainLayer object at 0x000001F00ECE9940>]
[[11.757146  -1.7759448 -7.3816853 ... -1.6559223 -1.6663467  0.       ]]
[[11.757146  -1.7759448 -7.3816853 ... -1.6559223 -1.6663467  0.       ]]
[[0. 0. 0. ... 0. 0. 0.]]
```

### Expected behavior

`load_tf_weights` should be able to load weights like `final_logits_bias`, and the TF checkpoint should be loaded correctly.

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Fixes #18802 

As stated in the issue above, `final_logits_bias` in `TFMarianMTModel` are not being loaded at `from_pretrained(...)` time. The PT model has this variable defined, and thus the outputs of the model in the two frameworks are very different (>1e-1).

Actually, these weights are also not being stored when the TF version is saved, for the same reason -- only layers are stored/loaded with the functions we are using (`.save_weights` and `.load_weights`), and this bias weight is not inside a layer.

As a solution, this PR moves the bias to a layer and creates an alias for it, resulting in no interface changes. After this change, the models from `Helsinki-NLP` can be converted with the `pt-to-tf` CLI, passing all the quality checks.

⚠️ Other models have this pattern, so I will apply the change to them in a separate PR if this one gets approved.

### CODE REVIEW

**Comment by ydshieh** [comment]
@gante Thanks a lot. It looks like it works well!

However, there is one thing I don't understand quite well.


```bash
(Pdb) [x.name for x in model.non_trainable_weights]
['final_logits_bias:0']
```
and this is good as it makes loading correctly. But I was thinking I will see `['final_logits_bias.final_logits_bias:0']`, as you pass the name to the layer as well as in `add_weight`.

Is it true that when we use `add_weight` inside a layer, that layer name won't appear in the variable name for that weight?

**Review by ydshieh — APPROVED** [review]
Looks good to me, as it works. But I left a question.
Thanks a lot @gante

**Comment by gante** [comment]
@ydshieh hah, I had the same question but I tried, it worked, and I forgot to dig deeper to understand why :D 

After some digging, I found that it is poorly documented -- variables created with `.add_weight` are set without any name scope, i.e. their name consists of the name set in `name`. This is opposed to the weights from layers, such as `tf.keras.layers.Dense`, that automatically get a scoped name according to the `name` of the layers (e.g. `foo/bar/weights:0`).

This implies that initializing `BiasLayer` with a `name` has no effect whatsoever regarding weight storing/loading. If we wanted the weights to have a scoped name (we don't here), we could either hardcode it in `name` or use `tf.name_scope`.

I'm adding a link to this comment in the code, for future reference.

**Comment by ydshieh** [comment]
Thanks a lot @gante , you are the best!

**Review by patrickvonplaten — APPROVED** [review]
Looks good to me!
